### PR TITLE
Rename provision config instance

### PR DESCRIPTION
**Pull request by cgchinmay**
### Description
[Describe what this change achieves]
renames provision_config_instance to cluster-config

### Issues Resolved
[List any issues this PR will resolve]
https://github.com/opensearch-project/opensearch-benchmark/issues/310

### Testing
- [ ] New functionality includes testing

[Describe how this change was tested]

validated with `make test` and `make it`

Running help with list command shows the renamed options
```
python3 benchmark.py list cluster-configs

Available cluster-configs:

Name                  Type            Description
--------------------  --------------  --------------------------------------
16gheap               cluster-config  Sets the Java heap to 16GB
1gheap                cluster-config  Sets the Java heap to 1GB
24gheap               cluster-config  Sets the Java heap to 24GB
2gheap                cluster-config  Sets the Java heap to 2GB
4gheap                cluster-config  Sets the Java heap to 4GB
8gheap                cluster-config  Sets the Java heap to 8GB
defaults              cluster-config  Sets the Java heap to 1GB
basic-license         mixin           Basic License
debug-non-safepoints  mixin           More accurate CPU profiles
ea                    mixin           Enables Java assertions
fp                    mixin           Preserves frame pointers
g1gc                  mixin           Enables the G1 garbage collector
parallelgc            mixin           Enables the Parallel garbage collector
trial-license         mixin           Trial License
unpooled              mixin           Enables Netty's unpooled allocator

-------------------------------
[INFO] SUCCESS (took 0 seconds)
-------------------------------

--------------------------------------------------
python3 benchmark.py list --help             
usage: benchmark.py list [-h] [--limit LIMIT] [--workload-repository WORKLOAD_REPOSITORY | --workload-path WORKLOAD_PATH] [--workload-revision WORKLOAD_REVISION]
                         [--distribution-version DISTRIBUTION_VERSION] [--cluster-config-path CLUSTER_CONFIG_PATH]
                         [--cluster-config-repository CLUSTER_CONFIG_REPOSITORY] [--cluster-config-revision CLUSTER_CONFIG_REVISION] [--quiet] [--offline]
                         configuration

positional arguments:
  configuration         The configuration for which Benchmark should show the available options. Possible values are: telemetry, workloads, pipelines,
                        test_executions, cluster-configs, opensearch-plugins

optional arguments:
  -h, --help            show this help message and exit
  --limit LIMIT         Limit the number of search results for recent test_executions (default: 10).
  --workload-repository WORKLOAD_REPOSITORY
                        Define the repository from where Benchmark will load workloads (default: default).
  --workload-path WORKLOAD_PATH
                        Define the path to a workload.
  --workload-revision WORKLOAD_REVISION
                        Define a specific revision in the workload repository that Benchmark should use.
  --distribution-version DISTRIBUTION_VERSION
                        Define the version of the OpenSearch distribution to download. Check https://opensearch.org/docs/version-history/ for released versions.
  --cluster-config-path CLUSTER_CONFIG_PATH
                        Define the path to the cluster-config and plugin configurations to use.
  --cluster-config-repository CLUSTER_CONFIG_REPOSITORY
                        Define repository from where Benchmark will load cluster-configs (default: default).
  --cluster-config-revision CLUSTER_CONFIG_REVISION
                        Define a specific revision in the cluster-configs repository that Benchmark should use.
  --quiet               Suppress as much as output as possible (default: false).
  --offline             Assume that Benchmark has no connection to the Internet (default: false).
```
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/benchmark.py [hunk 1]
```diff
@@ -440,14 +440,14 @@ def add_workload_source(subparser):
         "--test-procedure",
         help=f"Define the test_procedure to use. List possible test_procedures for workloads with `{PROGRAM_NAME} list workloads`.")
     test_execution_parser.add_argument(
-        "--provision-config-instance",
-        help=f"Define the provision_config_instance to use. List possible "
-        f"provision_config_instances with `{PROGRAM_NAME} list "
-        f"provision_config_instances` (default: defaults).",
+        "--cluster-config",
+        help=f"Define the cluster_config to use. List possible "
```
Comment (IanHoang) [inline osbenchmark/benchmark.py:444]: Nit: Update `cluster_config` --> `cluster-config` to keep consistency in help texts

File: osbenchmark/benchmark.py [hunk 2]
```diff
@@ -440,14 +440,14 @@ def add_workload_source(subparser):
...
+        f"cluster_configs with `{PROGRAM_NAME} list "
+        f"cluster_configs` (default: defaults).",
         default="defaults")  # optimized for local usage
     test_execution_parser.add_argument(
-        "--provision-config-instance-params",
-        help="Define a comma-separated list of key:value pairs that are injected verbatim as variables for the provision_config_instance.",
+        "--cluster-config-params",
+        help="Define a comma-separated list of key:value pairs that are injected verbatim as variables for the cluster_config.",
```
Comment (IanHoang) [inline osbenchmark/benchmark.py:450]: Nit: `cluster_config` --> `cluster-config`

File: osbenchmark/benchmark.py [hunk 3]
```diff
@@ -299,14 +299,14 @@ def add_workload_source(subparser):
              "Check https://opensearch.org/docs/version-history/ for released versions.",
         default="")
     install_parser.add_argument(
-        "--provision-config-instance",
-        help=f"Define the provision_config_instance to use. List possible "
-        f"provision_config_instances with `{PROGRAM_NAME} list "
-        f"provision_config_instances` (default: defaults).",
+        "--cluster-config",
+        help=f"Define the cluster_config to use. List possible "
```
Comment (IanHoang) [inline osbenchmark/benchmark.py:303]: Nit: `cluster_config` and `cluster_configs` --> `cluster-config` and `cluster-config`

File: osbenchmark/benchmark.py [hunk 4]
```diff
@@ -274,15 +274,15 @@ def add_workload_source(subparser):
...
+        help="Define the repository from where Benchmark will load provision_configs and cluster_configs (default: default).",
         default="default")
     install_parser.add_argument(
         "--provision-config-revision",
         help="Define a specific revision in the provision_config repository that Benchmark should use.",
         default=None)
     install_parser.add_argument(
         "--provision-config-path",
-        help="Define the path to the provision_config_instance and plugin configurations to use.")
+        help="Define the path to the cluster_config and plugin configurations to use.")
```
Comment (IanHoang) [inline osbenchmark/benchmark.py:285]: Nit: `cluster_config` --> `cluster-config`

**Review by IanHoang — CHANGES_REQUESTED** [review]
Left some minor suggestions but overall looks good to me! Were you able to test this? Could you provide a sample output of the help text and a brief run with a cluster-config to ensure that it works? 

When we merge this in, we'll merge this into a separate branch first since these changes are considered breaking-changes. Thus, we don't want to merge this into mainline yet until we're ready to release OSB 2.0.0

**Comment by cgchinmay** [comment]
Yes, I will make the suggested changes and share output

**Review by IanHoang — APPROVED** [review]

**Pull request merged**